server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -469,8 +485,10 @@ public void handleResponse(PeersResponse response) {
 
                         peersRequestInFlight = false;
 
-                        response.getMasterNode().map(DiscoveryNode::getAddress).ifPresent(PeerFinder.this::startProbe);
-                        response.getKnownPeers().stream().map(DiscoveryNode::getAddress).forEach(PeerFinder.this::startProbe);
+                        response.getMasterNode().ifPresent(node -> startProbe(node.getAddress()));
```

Comment:
Probably irrelevant but both of these showed in profiling and allocations under a lock shared with transport threads are just meh :)